core/auth/auth.go

```diff
@@ -51,36 +66,29 @@ func SetupAuth(cfg *config.Configuration) (rt http.RoundTripper, err error) {
 	}
 }
 
-// DefaultAuth will search for a valid token in several locations.
-// It will first try to find the token in the STACKIT_SERVICE_ACCOUNT_TOKEN. If not present, it will
-// check the credentials file located in STACKIT_CREDENTIALS_PATH, if specified, or in
-// $HOME/.stackit/credentials.json as a fallback. If the token is found, the TokenAuth flow is used and DefaultAuth returns
-// an http.RoundTripper that can be used to make authenticated requests.
+// DefaultAuth will search for a valid service account key or token in several locations.
+// It will first try to use the key flow, by looking into the variables STACKIT_SERVICE_ACCOUNT_KEY, STACKIT_SERVICE_ACCOUNT_KEY_PATH,
+// STACKIT_PRIVATE_KEY and STACKIT_PRIVATE_KEY_PATH. If the keys cannot be retrieved, it will check the credentials file located in STACKIT_CREDENTIALS_PATH, if specified, or in
+// $HOME/.stackit/credentials.json as a fallback. If the key are found and are valid, the KeyAuth flow is used.
+// If the key flow cannot be used, it will try to find a token in the STACKIT_SERVICE_ACCOUNT_TOKEN. If not present, it will
+// search in the credentials file. If the token is found, the TokenAuth flow is used.
+// DefaultAuth returns an http.RoundTripper that can be used to make authenticated requests.
 // In case the token is not found, DefaultAuth fails.
 func DefaultAuth(cfg *config.Configuration) (rt http.RoundTripper, err error) {
 	if cfg == nil {
 		cfg = &config.Configuration{}
 	}
 
-	var token string
-	// Check token
-	token, tokenSet := os.LookupEnv("STACKIT_SERVICE_ACCOUNT_TOKEN")
-	if !tokenSet || token == "" {
-		token, err = readTokenFromCredentialsFile(cfg.CredentialsFilePath)
+	// Key flow
+	rt, err = KeyAuth(cfg)
+	if err != nil {
+		keyFlowErr := err
+		// Token flow
+		rt, err = TokenAuth(cfg)
 		if err != nil {
-			return nil, fmt.Errorf("STACKIT_SERVICE_ACCOUNT_TOKEN not set. Trying to read from credentials file: %w", err)
+			return nil, fmt.Errorf("no valid credentials were found: trying key flow: %s, trying token flow: %w", keyFlowErr.Error(), err)
```

Comment:
Can you use "%w" instead of "%s", and have `keyFlowErr` instead of `keyFlowErr.Error()`?

Reply:
No, it throws a compilation error:
`auth/auth.go:89: fmt.Errorf call has more than one error-wrapping directive %w`